source/solvers/heat_transfer.cc

```diff
@@ -64,9 +64,19 @@ HeatTransfer<dim>::setup_assemblers()
 
   if (this->simulation_parameters.multiphysics.viscous_dissipation)
     {
-      this->assemblers.push_back(
-        std::make_shared<HeatTransferAssemblerViscousDissipation<dim>>(
-          this->simulation_control));
+      if (not this->simulation_parameters.multiphysics.VOF)
+        {
+          this->assemblers.push_back(
+            std::make_shared<HeatTransferAssemblerViscousDissipation<dim>>(
+              this->simulation_control));
+        }
+      else
+        {
+          // Call for the specialized assembler
+          this->assemblers.push_back(
+            std::make_shared<HeatTransferAssemblerViscousDissipationVOF<dim>>(
+              this->simulation_control));
+        }
```

Comment:
Yes I agree for readability, but for performances I always try to put the most probable case as the first `if` statement. This considered, do you still wish that I change the order? (I would be very ok with favoring readability also)

Reply:
It won't change performance. This place has no impact on performance at all because choosing the assembler is done once per iteration and these are cheap objects to create.